### DISCUSSION TITLE
Offer a store (container) or allow to add custom containers

### DISCUSSION THREAD

**Discussion by ghost** [nextcloud/all-in-one#360] [Ideas] [answered]
Offer a store. So that, like Nextcloud, everyone can publish apps in the store. Only that here everyone can publish a container. For example OSRM Server, Mail Server, ...

**Comment by szaimen** [comment]
Hi there 👋,

thats an interesting idea but no guarantee that it gets implemented like this. 

What I had in mind in the long run is allowing to provide an url that merges the container definition with that one: https://github.com/nextcloud/all-in-one/blob/main/php/containers.json. By doing so you would be able to add custom containers that of course need to be built and provided in the correct way.

**Comment by szaimen** [comment]
Just to note down my ideas:

it could work like this:
we could allow mounting a file into the mastercontainer which would automatically be merged into the containers definition. 
Or maybe allow to enable this using a switch in the gui which loads it automatically? Question is how the file can then be adjusted
This file would be needed to be edited by hand which would allow to customize certain values.
Of course the used containers and the template would need to be built and provided to match the custom container labeling.
There could be a check that is recommended to be run (in the container?) before starting as it would kill the container.
Thinkable docker containers are: (Basically web-services make sense here.)
```
caddy reverse proxy (with geoblocking?)
vaultwarden
plex and/or jellyfin
pi-hole
pivpn/wireguard (with webinterface?)
```

These should then be hosted under a different org and also get a org on docker hub. Builsing would then be done in the repo which impepve the trust problem.

drive management like mounting and smart would need to still be done on the host. 

Later we could think about also putting remotedesktop and programs and backup viewer, smb-server, ddclient etc. into a container... (so that really basically everything is in a container... except of course drive mounting, formatting, smart...)

Or think about working on the vm to include aio and simply add all things on top that aio cannot provide?

---

As requisite it needs https://github.com/nextcloud/all-in-one/issues/695

**Comment by szaimen** [comment, marked as answer]
After thinking a bit more about this, I don't think we will add this as it will make AIO much more complex and thus harder to maintain. 

Also, adding separate docker containers should be easy enough in most cases.

**Reply by ghost** [reply]
Possibly it is helpful can this be included in the documentation?
